deepmd/pt/model/task/fitting.py

```
@@ -409,6 +413,48 @@ def _net_out_dim(self):
+                originating from the `i`-th data system.
+            - Callable[[], list[dict]]: A lazy function that returns data samples in the above format
+                only when needed. Since the sampling process can be slow and memory-intensive,
+                the lazy function helps by only sampling once.
+        path : Optional[DPPath]
+            The path to the stat file.
+
+        """
+        if callable(merged):
+            sampled = merged()
+        else:
+            sampled = merged
```

Ruff (0.8.2)

Use ternary operator `sampled = merged() if callable(merged) else merged` instead of `if`-`else`-block

Replace `if`-`else`-block with `sampled = merged() if callable(merged) else merged`

(SIM108)